kotest-assertions/kotest-assertions-konform/src/jvmMain/kotlin/io/kotest/assertions/konform/matchers.kt

```diff
@@ -0,0 +1,27 @@
+package io.kotest.assertions.konform
+
+import io.konform.validation.Invalid
+import io.konform.validation.Valid
+import io.konform.validation.Validation
+import io.kotest.Matcher
+import io.kotest.MatcherResult
+import io.kotest.should
+
+infix fun <T> Validation<T>.shouldBeValid(value: T) = this should beValid(value)
+
+fun <A> beValid(a: A) = object : Matcher<Validation<A>> {
+   override fun test(value: Validation<A>): MatcherResult =
+      MatcherResult(value(a) is Valid, "$a should be valid", "$a should not be valid")
```

Comment:
Maybe we could include the validation failure in the messsage

Reply:
Done. Since errors is a private field of Invalid, I've added a print of the whole object